sdk/go/common/resource/plugin/langruntime_plugin.go

```diff
@@ -27,10 +30,17 @@ import (
 	"github.com/pulumi/pulumi/sdk/v2/go/common/util/contract"
 	"github.com/pulumi/pulumi/sdk/v2/go/common/util/logging"
 	"github.com/pulumi/pulumi/sdk/v2/go/common/util/rpcutil/rpcerror"
+	"github.com/pulumi/pulumi/sdk/v2/go/common/version"
 	"github.com/pulumi/pulumi/sdk/v2/go/common/workspace"
 	pulumirpc "github.com/pulumi/pulumi/sdk/v2/proto/go"
 )
 
+// HostOnlyRuntimeMode can be used to launch the engine in "host"-only mode. That is to say, the engine and
+// resource monitors will be spawned, but no language runtime will be created. This allows automated
+// scenarios that drive the engine from outside of the engine itself, such as, for example, a Node.js
+// program spawning, deploying, and tearing down the engine's resources.
+const HostOnlyRuntimeMode = "host"
```

Comment:
That's a great idea. I've been worried that the host lifecycle being decoupled from the execution of the program could lead to leaked hosts and updates that don't get closed when the program encounters an exception. 

I've set up time for us to discuss design in more detail next week, and opened the following issue to track: https://github.com/pulumi/pulumi/issues/5249

It would be my preference to not design it in this PR though. To that end, I've hidden the host command via https://github.com/pulumi/pulumi/pull/4977/commits/aba397a8c8e0f760040abc9abe57392aeab02bf3. With the host command hidden, and the fact that the underlying implementation of host mode isn't exposed to users via automation API (just the presence of workspace.program is all), I think we should be able to swap this out without any user-facing impact. 

Can we merge this with host mode as is? This allows @metral and others to start contributing while I work on this in parallel.